client/src/BuyerTenderList.js

```diff
@@ -1,27 +1,57 @@
-import { useEffect, useState } from "react";
+import React, { useState, useEffect } from "react";
+import { useParams, useNavigate } from "react-router-dom";
 import { get } from "./TenderClient";
 
 const BuyerTenderList = () => {
-	const [loading, setLoading] = useState(true);
+	const { pageNumber } = useParams();
+	const currentPage = pageNumber ? parseInt(pageNumber, 10) : 1;
 	const [buyerTenders, setBuyerTenders] = useState([]);
 	const [errorMsg, setErrorMsg] = useState(null);
+	const [loading, setLoading] = useState(true);
+	const [pagination, setPagination] = useState({
+		itemsPerPage: 5,
+		currentPage: currentPage,
+		totalPages: 1,
+	});
+	const navigate = useNavigate();
 
 	function dateFormat(date) {
 		return date ? date.split("T")[0] : "N/A";
 	}
 
-	useEffect(() => {
-		const fetchBuyerTenders = async () => {
-			try {
-				const data = await get("api/buyer-tender?page=1");
-				setLoading(false);
+	const fetchTenders = async (page) => {
+		setLoading(true);
+		try {
+			const data = await get(`/api/buyer-tender?page=${page}`);
+			if (data && data.results && data.pagination) {
```

Comment:
This check is not necessary. If we get a `200`, we know all this will be present.